library/src/main/java/com/cube/storm/ui/lib/adapter/StormSpotlightAdapter.java

```diff
@@ -95,10 +98,31 @@ public Object instantiateItem(@NonNull ViewGroup container, int position)
 		final SpotlightImageProperty spotlightItem = getItem(position);
 		if (spotlightItem != null)
 		{
+			View textContainer = view.findViewById(R.id.text_container);
 			ImageView imageView = view.findViewById(R.id.image_view);
-			TextView textView = view.findViewById(R.id.text_ticker);
+			TextView category = view.findViewById(R.id.category);
+			TextView title = view.findViewById(R.id.title);
+			TextView description = view.findViewById(R.id.description);
+
 			imageView.populate(spotlightItem.getImage());
-			textView.populate(spotlightItem.getText(), spotlightItem.getLink());
+			category.populate(spotlightItem.getCategory());
+			title.populate(spotlightItem.getTitle());
+			description.populate(spotlightItem.getDescription());
+
+			// Hide text container under spotlight image if no text to populate
+			Processor<TextProperty, String> textProcessor = UiSettings.getInstance().getTextProcessor();
+			String categoryString = textProcessor.process(spotlightItem.getCategory());
+			String titleString = textProcessor.process(spotlightItem.getTitle());
+			String descriptionString = textProcessor.process(spotlightItem.getDescription());
+			if (TextUtils.isEmpty(categoryString) && TextUtils.isEmpty(titleString) && TextUtils.isEmpty(descriptionString))
+			{
+				textContainer.setVisibility(View.GONE);
```

Comment:
The views populated by ViewHolders are recycled, meaning the same view could be given a different model and re-populated. However, your call to setVisibility here will set the textContainer to GONE indefinitely, even if it is subsequently given a `spotlightItem` that does not have missing text.

The solution is to add an `else { textContainer.setVisibility(View.VISIBLE) }`